src/test/consensus/Validations_test.cpp

```diff
@@ -1080,6 +1080,14 @@ class Validations_test : public beast::unit_test::suite
         }
     }
 
+    void
+    testDifferentCookies()
+    {
+        testcase("Multiple Validations");
```

Comment:
I think it would be good to have more unit tests here. In addition, I would ask @manojsdoshi to consider to have automated tests for this PR to force the logic to execute on QE network. This should never happen naturally on the real network but it would be useful to see it in action in test.